### ISSUE TITLE
ccdb command ignores variation parent

### ISSUE THREAD

**Issue by DraTeots** [JeffersonLab/ccdb#5]
As we discussed yesterday in the BCAL reconstruction meeting, we had 
 decided for the variation=calib to duplicate some of the constants 
 (e.g. ADC_gains) from the variation=default. According to Mark Ito, a 
 variation will default to the parent data if no specific 'variation 
 data' has been loaded. This would save us considerable effort in 
 completing the variation=calib. However, this does not seem to work:

```
ifarm1102> ccdb dump /BCAL/ADC_gains::calib > ! ADC_gains.txt
ifarm1102> more ADC_gains.txt
There is no data for table /BCAL/ADC_gains, run 0, variation 'calib'
Cannot fill data for assignment with this ID
ifarm1102> ccdb dump /BCAL/ADC_gains::default > ! ADC_gains.txt
ifarm1102> more ADC_gains.txt

 #
   3.32441e-05
   6.81519e-05
   4.44338e-05
   8.16211e-05
   3.15279e-05
   5.12456e-05
   5.01645e-05
   5.9035e-05
 ........

```

**Comment by sdobbs** [comment]
FYI, having the proper fallback would be very useful in the calibration management scripts I'm working on, which use the python API and several different variations.

**Comment by DraTeots** [comment]
I actually have fixed that. But it was done outside of python API you are using (directly in can/dump commands). It was done this way to control if we need to fallback or not. 

Now I consider it was a wrong decision and fallback should be made in get_assigment(...) function.
Fixing it.

**Comment by DraTeots** [comment]
Fixed in
9dc19053cae143d215d2561156bc24a1866656c6

**Comment by DraTeots** [comment]
Indeed. The issue is still in cat command. The problem is that falling back to variation parent is implemented in get_assignment method, but it happens that cat uses get_assignment_s_

Reviewing both method I realized, that a signature of `get_assignment` has to be changed. I hope it will not break your code. 

After the update get_assignment will be:

``` python
def get_assignment(self, path_or_table, run, variation, date_and_time=None, use_variation_tree=True):
 """
        Gets the latest assignment that matches the parameters

        @param run: run number
        @param path_or_table: string with name path of table or Table object
        @param variation: name of the variation or variation object
        @param date_and_time: Constants should be of this or later date
        @param use_variation_tree:

        @return: latest assignment that matches the parameters
        @rtype: Assignment

        """
```

Probably I have to also add `get_assignment_by_request` function too.

**Comment by DraTeots** [comment]
Fixed in 1a045733276030780f4e47117a794ca94f324982
Needs more tests
